### PR TITLE
Add rollup for module release

### PR DESCRIPTION
**Pull request by subdavis**
Finally some progress toward an esm distributable of `vue-media-annotator`.

Apparently if the root import of your `.vue` files is a `.js` file, rollup-plugin-vue just builds the vue SFCs as if they were normal javascript.  You can't get type definitions, but you can at least build.

I've broken the library into 2 import roots: 1 for typescript with defs, 1 for components without.  Here's how you use it:

``` js
import { VideoAnnotator, LayerManager } from 'vue-media-annotator/dist/components';
import { providers, use } from 'vue-media-annotator';
```

I've confirmed with a consumer app that this works out-of-the-box in a regular vue app.

### CODE REVIEW

**Comment by subdavis** [comment]
Released `1.0.0-beta.1` for testing.

File: client/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+export {
+  listUtils,
+  providers,
+  Track,
+  utils,
+
+  use,
+  layers,
+  // components,
```
Comment (subdavis) [inline client/src/index.ts:18]:
```suggestion
```

File: client/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import * as listUtils from './listUtils';
+import * as providers from './provides';
+import Track from './track';
+import * as utils from './utils';
+
+import * as use from './use';
+import * as layers from './layers';
+// import * as components from './components';
```
Comment (subdavis) [inline client/src/index.ts:8]:
```suggestion
```

File: client/src/components/TrackItem.vue [hunk 3]
```diff
@@ -321,7 +321,7 @@ export default defineComponent({
     }
   }
   .input-box {
-    border: 1px solid rgb(255, 255, 255, 0.15);
+    border: 1px solid rgba(255, 255, 255, 0.15);
```
Comment (AlmightyYakob) [inline client/src/components/TrackItem.vue:324]: Seems unrelated?

File: client/app/compositionplugin.ts [hunk 4]
```diff
@@ -1,6 +0,0 @@
-import Vue from 'vue';
-import VueCompositionApi from '@vue/composition-api';
-
-Vue.use(VueCompositionApi);
-
-export default VueCompositionApi;
```
Comment (AlmightyYakob) [inline client/app/compositionplugin.ts:6]: Why was this file removed?

File: client/rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+const clean = true;
+const tsconfig = 'tsconfig.rollup.json';
+const external = [
+  '@vue/composition-api',
+  '@flatten-js/interval-tree',
+  'd3',
+  'geojs',
+  'lodash',
+  'vue',
+];
```
Comment (AlmightyYakob) [inline client/rollup.config.js:18]: Is this ultimately specified to reduce bundle size?

**Review by AlmightyYakob — COMMENTED** [review]
Just a few comments, but generally looks good to me.

File: client/src/components/TrackItem.vue (hunk 3 above)
Comment (subdavis) [reply client/src/components/TrackItem.vue:324]: It was breaking build because the rollup sass plugin validates sass syntax.  Nothing in the webpack build was doing that.

File: client/app/compositionplugin.ts (hunk 4 above)
Comment (subdavis) [reply client/app/compositionplugin.ts:6]: It was never being imported.

File: client/rollup.config.js (hunk 5 above)
Comment (subdavis) [reply client/rollup.config.js:18]: It does that, but only because there's just no other option.  Otherwise you run into the issue where multiple instances of Vue (and every other dependency) are instantiated and nothing works.

The point is to defer dependency resolution to the consumer.

**Comment by AlmightyYakob** [comment]
Of note, if you're not running the same (at least major?) version of `@vue/composition-api` as this is (`1.0.0-beta.14`), you get composition api errors down the call stack (I got mine in `useTrackStore` for example). Besides that, it seems to work fine. I have to update my app to use `provideAnnotator`, but that seems separate from this.

**Review by AlmightyYakob — APPROVED** [review]
